app/javascript/packs/reducers/cuisines/index.js

```diff
@@ -0,0 +1,28 @@
+import { createReducer } from '../../modules/redux';
+import { normalizeArrayData} from '../../modules/redux/helpers';
+import { ACTIONS_NAME } from './actions';
+
+const initState = {
+    cuisines: {},
+    isLoaded: false,
+    isLoading: false,
+};
+
+export default createReducer({
+    [ACTIONS_NAME.FETCH_CUISINES_STARTED]: (state) => ({
+        ...state,
+        isLoaded: false,
+        isLoading: true,
+    }),
+    [ACTIONS_NAME.FETCH_CUISINES_FAILED]: (state) => ({
+        ...state,
+        isLoaded: true,
+        isLoading: false,
+    }),
+    [ACTIONS_NAME.FETCH_CUISINES_DONE]: (state, action) => ({
+        ...state,
+        cuisines: normalizeArrayData(action.cuisines),
+        isLoaded: true,
+        isLoading: false,
```

Comment:
I once got to a conclusion it's better to use a `status` enum (`EMPTY`, `LOADING`, `LOADED`, etc.) instead of multiple booleans.

Reply:
ye, I never sure which one is better, I tend to agree